src/neo-vm/Types/PrimitiveType.cs

```diff
@@ -6,52 +6,53 @@ namespace Neo.VM.Types
 {
     public abstract class PrimitiveType : StackItem
     {
-        public override bool Equals(StackItem other)
+        public abstract ReadOnlyMemory<byte> Memory { get; }
+        public virtual int Size => Memory.Length;
+        public virtual ReadOnlySpan<byte> Span => Memory.Span;
+
+        public sealed override bool Equals(object obj)
         {
-            if (ReferenceEquals(this, other)) return true;
-            if (other is null) return false;
-            if (!(other is PrimitiveType p)) return false;
-            return Unsafe.MemoryEquals(ToByteArray(), p.ToByteArray());
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj is PrimitiveType p) return Equals(p);
+            return false;
         }
 
-        public virtual int GetByteLength()
+        public virtual bool Equals(PrimitiveType other)
         {
-            return ToByteArray().Length;
+            if (ReferenceEquals(this, other)) return true;
```

Comment:
This line could be called twice. L15 and here

Reply:
But this method may also be called separately.